File: cotgenerator/src/main/java/com/jon/cotgenerator/service/GeneratorService.java
package com.jon.cotgenerator.service;

import com.jon.common.service.CotService;
import com.jon.common.utils.Key;
import com.jon.common.utils.PrefUtils;

public class GeneratorService extends CotService {
    @Override
    public void onCreate() {
        super.onCreate();

        /* Only initialise the GPS requests if the option is enabled in settings */
        if (PrefUtils.getBoolean(prefs, Key.FOLLOW_GPS_LOCATION)) {
            initialiseFusedLocationClient();
        }
    }
}

File: cotgenerator/src/main/java/com/jon/cotgenerator/GeneratorService.kt
package com.jon.cotgenerator

import com.jon.common.prefs.getBooleanFromPair
import com.jon.common.service.CotService

class GeneratorService : CotService() {
    override fun onCreate() {
        super.onCreate()

        /* Only initialise the GPS requests if the option is enabled in settings */
        if (prefs.getBooleanFromPair(GeneratorPrefs.FOLLOW_GPS_LOCATION)) {
            initialiseFusedLocationClient()
        }
    }
}
